src/blocks/ImageBlock/Image.client.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * A custom HTML element that enhances an `<img>` element with additional behavior.
+ * Automatically removes the inline `background-image` style once the image is fully loaded.
```

Comment:
I would remove all other comments except for these top ones. I understand you saw the `Tabs` component as a reference. That one was based on an article on web.dev that already included all the comments. Since this script is much more straightforward, I'd remove the rest of the comments.
The only thing I would add, is the explanation why. So:

```ts
/**
 * A custom HTML element that enhances an `<img>` element with additional behaviour to remove the inline `background-image` style once the image is fully loaded. This prevents blurred placeholder backgrounds from still being visible once an image with transparency (like png or webp) is loaded.
```